### PR TITLE
feat: drive subsys to max timeout reload

### PR DESCRIPTION
**Pull request by jiuker**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

feat: add drive max timeout reload

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/config-current.go [hunk 1]
```diff
@@ -97,6 +99,10 @@ func initHelp() {
 			Description: "enable callhome to MinIO SUBNET",
 			Optional:    true,
 		},
+		config.HelpKV{
+			Key:         config.DriveSubSys,
+			Description: "enable drive config",
```
Comment (harshavardhana) [inline cmd/config-current.go]:
```suggestion
			Description: "enable drive specific settings",
```

File: go.mod [hunk 2]
```diff
@@ -247,3 +247,5 @@ require (
 	gopkg.in/square/go-jose.v2 v2.6.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/minio/madmin-go/v3 => github.com/jiuker/madmin-go/v3 v3.0.0-20231122072459-e88ab8ab6645
```
Comment (harshavardhana) [inline go.mod]:
```suggestion
replace github.com/minio/madmin-go/v3 => github.com/jiuker/madmin-go/v3 v3.0.0-20231122072459-e88ab8ab6645
```

Please use v3.0.33

File: internal/config/drive/help.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+var (
+	// MaxTimeout is the max timeout for drive
+	MaxTimeout = "max_timeout"
+	// HelpDrive is help for drive
+	HelpDrive = config.HelpKVS{
+		config.HelpKV{
+			Key:         MaxTimeout,
+			Type:        "string",
+			Description: "set max_timeout for drive, default 2min",
```
Comment (harshavardhana) [inline internal/config/drive/help.go]:
```suggestion
			Description: "set per call max_timeout for the drive, defaults to 2 minutes",
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

File: internal/config/drive/drive.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	},
+}
+
+// Config represents the subnet related configuration
+type Config struct {
+	// Flag indicating whether callhome is enabled.
+	MaxTimeout time.Duration `json:"maxTimeout"`
+}
+
+var mutex sync.RWMutex
```
Comment (klauspost) [inline internal/config/drive/drive.go]: This seems misplaced.

Add the mutex to the data it is protecting. Ie inside the Config. Package level locks are a no-no.

**Review by klauspost — CHANGES_REQUESTED** [review]

**Review by klauspost — APPROVED** [review]

**Pull request merged**